ethexe/common/src/db.rs

```diff
@@ -81,6 +81,7 @@ pub trait CodesStorage: Send + Sync {
 
     fn program_code_id(&self, program_id: ProgramId) -> Option<CodeId>;
     fn set_program_code_id(&self, program_id: ProgramId, code_id: CodeId);
+    fn program_ids(&self) -> BTreeSet<ProgramId>;
```

Comment:
It seems that you could just iter through program_id to code_id, verifying both K and V on eth ?

Reply:
tbd later